### PR TITLE
fix(AppHeaderCatalogDropdown): axe fixes

### PR DESCRIPTION
**Pull request by zippyzow**
### Overview

Axe Bug 1: 
[A] Screen Reader: Headings in Catalog menu are not marked up semantically as such
http://codecademy.test.io/products/9763/test_cycles/98728/bugs/1945138

Axe Bug 2:
[A] Screen reader:The title of the first section is missed and not announced
http://codecademy.test.io/products/9763/test_cycles/98728/bugs/1944593

Axe Bug 3: 
[A] Screen Reader: Related links are not grouped in a list
http://codecademy.test.io/products/9763/test_cycles/98728/bugs/1945160

Test that each of these issues have been remedied. 

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: DISC-612, DISC-613, DISC-614
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/30495/ 'Named link title')  | [Monolith Env](https://pr-30495-monolith.dev-eks.codecademy.com/ 'Named link title') |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/portal-app/pull/1888 'Named link title')  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-1888/ 'Named link title')   |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.116-alpha.4d4615.0<br/>@codecademy/gamut-labs@31.11.5-alpha.4d4615.0<br/>@codecademy/styleguide@57.0.2-alpha.4d4615.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6269d78e1c9ee625fb477d26--gamut-preview.netlify.app>https://6269d78e1c9ee625fb477d26--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6269d78e1c9ee625fb477d26>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.116-alpha.a0f9ed.0<br/>@codecademy/gamut-labs@31.11.5-alpha.a0f9ed.0<br/>@codecademy/styleguide@57.0.2-alpha.a0f9ed.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6269f9af854089381ff55bd3--gamut-preview.netlify.app>https://6269f9af854089381ff55bd3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6269f9af854089381ff55bd3>Deploy Logs</a></p>

**Review by borisonr — APPROVED** [review]
lgtm!

File: packages/gamut-labs/src/AppHeader/AppHeaderElements/AppHeaderCatalogSection/index.tsx [hunk 1]
```diff
@@ -120,12 +123,19 @@ export const AppHeaderCatalogSection = React.forwardRef<
...
+                      data-focusablecatalog="true"
+                      as="h3"
+                      key={item.id}
+                      minWidth="12rem"
+                      tabIndex={-1}
+                    >
                       {item.text}
                     </StyledSubheader>
                   ) : (
                     <StyledAnchorBox key={item.id} minWidth="12rem">
```
Comment (vonbarown) [inline packages/gamut-labs/src/AppHeader/AppHeaderElements/AppHeaderCatalogSection/index.tsx:136]: when I get to a menu item link "Full-stack engineer" my focus jumps out back to the logo

Comment (zippyzow) [reply packages/gamut-labs/src/AppHeader/AppHeaderElements/AppHeaderCatalogSection/index.tsx:136]: yeah ive seen this behavior too (on prod as well). I find this flow has worked: tab to the logo, then I SHIFT right arrow to Catalog, CTRL+OPT+SPACE to get the dropdown, and then use up and down arrow to navigate through items. Does this work for you?

Comment (vonbarown) [reply packages/gamut-labs/src/AppHeader/AppHeaderElements/AppHeaderCatalogSection/index.tsx:136]: that worked for me

**Review by vonbarown — APPROVED** [review]

**Review by dreamwasp — APPROVED** [review]
all looks good to me + looks 💯  in both of the testing envs

**Pull request merged**